jujugui/static/gui/src/app/user/user.js

```diff
@@ -30,6 +41,23 @@ const User = class User {
     this.sessionStorage = cfg.sessionStorage || sessionStorage;
     this.localStorage = cfg.localStorage || localStorage;
     this._external = cfg.externalAuth || null;
+    this.expiration = cfg.expiration || getExpirationDate();
+  }
+
+  get expiration() {
+    return new Date(this.sessionStorage.getItem('expiration'));
+  }
+
+  set expiration(val) {
+    this.sessionStorage.setItem('expiration', val);
+  }
+
+  _checkExpiration() {
+    const now = new Date();
+    if (now > this.expiration) {
+      this.localStorage.clear();
+      this.sessionStorage.clear();
```

Comment:
This implies that we have the same expiration timeout for both the services (charm store, omni etc) and the juju controller. Should we keep the controller/model credentials? Or maybe not, because after 24 hours the storefront session would expire anyway? In any case, please add a comment.